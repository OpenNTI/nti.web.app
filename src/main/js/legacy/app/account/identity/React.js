const Ext = require('@nti/extjs');
const { getService } = require('@nti/web-client');
const { Identity } = require('@nti/web-profiles');
const LoginActions = require('internal/legacy/login/Actions');
require('internal/legacy/overrides/ReactHarness');

const AccountActions = require('../Actions');

module.exports = exports = Ext.define(
	'NextThought.app.account.identity.React',
	{
		extend: 'NextThought.ReactHarness',
		alias: 'widget.identity-react',
		component: Identity.Session,

		setTheme(scope, knockout) {},

		initComponent: function () {
			this.callParent(arguments);
			Object.assign(this.initialConfig, {
				addHistory: true,
				baseroute: '/app',
			});
			this.AccountActions = AccountActions.create();
			this.LoginActions = LoginActions.create();
		},

		async afterRender() {
			this.callParent(arguments);
			const service = await getService();
			const supportLinks = service.getSupportLinks();

			this.mon(Ext.getBody(), {
				click: this.onBodyClicks.bind(this, supportLinks),
			});
		},

		onBodyClicks(supportLinks, e) {
			let action = e.getTarget('a[data-action]')?.dataset.action;
			const link = e.getTarget('a[href]');
			const href = link?.getAttribute('href');

			// This is used to launch support logic the same way everywhere
			// we have a link that goes to the supportContact url. (so we can
			// just emit that url in the dom and not worry about coping logic)
			if (
				supportLinks.supportContact === href &&
				supportLinks.internalSupport
			) {
				action = action || 'support';
			}

			if (action) {
				if (call(this, `do-${action}`, link, e) !== false) {
					e.stopEvent();
				}
			}
		},

		showLink(target) {
			target.setAttribute('target', '_blank');
			return false;
		},

		'do-welcomeGuide'(target) {
			this.AccountActions.showWelcomePage(target.href);
		},

		'do-about': 'showLink',

		'do-privacy'(target) {
			this.AccountActions.showPrivacy(target);
		},

		'do-privacyForMinors'(target) {
			this.AccountActions.showChildrensPrivacy(target.href);
		},

		'do-terms'(target) {
			this.AccountActions.showTermsOfService(target.href);
		},

		'do-support'(target) {
			this.AccountActions.showContactUs();
		},

		'do-helpSite': 'showLink',

		'do-impersonate'() {
			this.LoginActions.handleImpersonate();
		},

		'do-logout'() {
			this.LoginActions.handleLogout();
		},
	}
);

function call(scope, action, ...args) {
	if (typeof action === 'string') {
		return call(scope, scope[action], ...args);
	}

	return action.apply(scope, args);
}
