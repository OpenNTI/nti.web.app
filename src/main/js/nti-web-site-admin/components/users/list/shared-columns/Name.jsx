
import { scoped } from '@nti/lib-locale';
import {
	DisplayName,
	Avatar,
	Placeholder,
	Typography,
	Variant,
} from '@nti/web-core';
import { LinkTo } from '@nti/web-routing';

import { VerticallyCentered } from './Common';

const t = scoped(
	'nti-web-site-admin.components.users.list.shared-columns.Name',
	{
		title: 'Name',
	}
);

const Info = styled.div`
	display: flex;
	flex-direction: column;
	align-items: flex-start;
	justify-content: center;
	flex: 1 1 auto;
	min-height: 60px;
	max-width: calc(100% - 75px);
	overflow: hidden;
`;

NameColumn.Name = t('title');
NameColumn.SortOn = 'alias';

NameColumn.CSSClassName = css`
	width: 350px;
`;
NameColumn.Placeholder = () => (
	<VerticallyCentered>
		<Avatar.Placeholder circular mr="lg" />
		<Info>
			<Placeholder.Text as="div" type="body" text="Placeholder" mb="xs" />
			<Placeholder.Text as="div" type="body" text="test.email@mail.com" />
		</Info>
	</VerticallyCentered>
);

NameColumn.Create = props => Variant(NameColumn, props);

export function NameColumn({ item, getUser = x => x, context }) {
	const user = getUser(item);

	return (
		<VerticallyCentered as={LinkTo.Object} object={user} context={context}>
			<Avatar entity={user} circular mr="lg" />
			<Info>
				<Typography type="body" as="div" color="dark" limitLines={1}>
					<DisplayName entity={user} as="span" />
				</Typography>
				{user.email && (
					<Typography type="body" as="div">
						{user.email}
					</Typography>
				)}
			</Info>
		</VerticallyCentered>
	);
}
