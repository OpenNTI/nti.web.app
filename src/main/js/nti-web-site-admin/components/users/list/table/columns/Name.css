.name-column {
	/* placeholder */
}

.cell {
	display: flex;
	align-items: center;
	justify-content: flex-start;
}

.cell svg,
.cell img {
	width: 40px;
	height: 40px;
	border-radius: 20px;
	margin-right: 20px;
}

.user-info {
	composes: veridically-center from './column.css';
	overflow: hidden;
	text-overflow: ellipsis;
	max-width: calc(100% - 150px);
}

.username {
	font-size: 14px;
	line-height: 19px;
	color: var(--primary-grey);
	white-space: nowrap;
}

.email {
	font-size: 14px;
	line-height: 19px;
	color: var(--secondary-grey);
	white-space: nowrap;
	text-overflow: ellipsis;
	overflow: hidden;
}
