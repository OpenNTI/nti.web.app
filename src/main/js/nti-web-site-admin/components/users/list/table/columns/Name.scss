@import '~@nti/style-common';

.users-table-container {
	.name-col {
		.cell {
			display: flex;

			svg, img {
				width: 40px;
				height: 40px;
				border-radius: 20px;
				margin-right: 20px;
				margin-top: 10px;
			}

			.user-info {
				margin-top: 10px;
				overflow: hidden;
				text-overflow: ellipsis;
				max-width: calc(100% - 150px);

				.username {
					font-size: 14px;
					line-height: 19px;
					color: var(--primary-grey);
					white-space: nowrap;
				}

				.email {
					font-size: 14px;
					line-height: 19px;
					color: $secondary-grey;
					white-space: nowrap;
					text-overflow: ellipsis;
					overflow: hidden;
				}
			}
		}
	}
}
