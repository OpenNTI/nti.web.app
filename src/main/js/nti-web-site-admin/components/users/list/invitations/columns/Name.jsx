import React from 'react';

import { scoped } from '@nti/lib-locale';
import { Avatar, Typography, Placeholder } from '@nti/web-core';

import { VerticallyCentered } from './Common';

const isAdminInvite = RegExp.prototype.test.bind(/siteadmininvitation/);

const t = scoped(
	'nti-web-site-admin.components.users.list.invitations.columns.Name',
	{
		title: 'Name',
		administrator: 'Administrator',
		learner: 'Learner',
	}
);

const Info = styled.div`
	display: flex;
	flex-direction: column;
	align-items: flex-start;
	justify-content: center;
	flex: 1 1 auto;
	min-height: 60px;
`;

Name.CSSClassName = css`
	width: 350px;
`;
Name.Name = t('title');
Name.SortOn = 'receiver';
Name.Placeholder = () => (
	<VerticallyCentered>
		<Avatar.Placeholder circular mr="lg" />
		<Info>
			<Placeholder.Text
				as="div"
				typography="body"
				text="thelongplaceholderlong@email.com"
				mb="xs"
			/>
			<Placeholder.Text as="div" typography="body" text="Administrator" />
		</Info>
	</VerticallyCentered>
);
export function Name({ item }) {
	const entity = {
		Username: item.receiver,
		initials: item.receiver.charAt(0),
	};

	return (
		<VerticallyCentered>
			<Avatar entity={entity} circular mr="lg" />
			<Info>
				<Typography
					as="div"
					type="body"
					color="dark"
					limitLines={1}
					data-test-id="receiver"
				>
					{item.receiver}
				</Typography>
				<Typography
					as="div"
					type="body"
					limitLines={1}
					data-test-id="type"
				>
					{isAdminInvite(item.MimeType)
						? t('administrator')
						: t('learner')}
				</Typography>
			</Info>
		</VerticallyCentered>
	);
}
