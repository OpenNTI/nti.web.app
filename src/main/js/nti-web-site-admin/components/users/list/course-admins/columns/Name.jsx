import { NameColumn } from '../../shared-columns/Name';

export const Name = NameColumn.Create({
	getUser: i => i.user,
	context: 'site-admin.course-admins-list-item',
});
