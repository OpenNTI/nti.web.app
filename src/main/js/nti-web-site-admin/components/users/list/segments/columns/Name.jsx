import React from 'react';

import { scoped } from '@nti/lib-locale';
import { Typography, Placeholder } from '@nti/web-core';
import { LinkTo } from '@nti/web-routing';

import { VerticallyCentered } from '../../shared-columns/Common';

const t = scoped(
	'nti-web-site-admin.components.users.list.segments.columns.Name',
	{
		title: 'Name',
	}
);

const Container = styled(VerticallyCentered)`
	padding: 21px 0.25rem 21px 30px;
`;

Name.CSSClassName = css`
	width: 350px;
	padding-left: 30px;
`;

Name.Name = t('title');
Name.SortOn = 'title';

Name.Placeholder = () => (
	<Container>
		<Placeholder.Text type="body" text="Segment Title" />
	</Container>
);

export function Name({ item }) {
	return (
		<Container as={LinkTo.Object} object={item}>
			<Typography
				type="body"
				color="dark"
				limitLines={1}
				data-testid="segment-title"
			>
				{item.title}
			</Typography>
		</Container>
	);
}
