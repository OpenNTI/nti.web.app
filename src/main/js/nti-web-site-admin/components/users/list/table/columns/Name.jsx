import './Name.scss';
import React from 'react';
import PropTypes from 'prop-types';
import {scoped} from '@nti/lib-locale';
import {DisplayName, Avatar} from  '@nti/web-commons';
import {LinkTo} from '@nti/web-routing';
import cx from 'classnames';

const t = scoped('nti-web-site-admin.components.users.list.table.columns.Name', {
	headerTitle: 'Type',
	title: 'Name'
});

export default class Name extends React.Component {
	static propTypes = {
		item: PropTypes.object.isRequired,
		store: PropTypes.object.isRequired
	}

	static cssClassName = 'name-col';

	static Name = () => t('title')

	static SortKey = 'alias'

	render () {
		const {item, store} = this.props;

		const type = store.filter === 'learners' ? 'users' : 'admins';
		const context = `site-admin.${type}-list-item`;

		return (
			<LinkTo.Object object={item} context={context} exact>
				<div className={cx('cell')}>
					<Avatar entity={item}/>
					<div className="user-info">
						<DisplayName entity={item}/>
						<div className="email">{item.email}</div>
					</div>
				</div>
			</LinkTo.Object>
		);
	}
}
