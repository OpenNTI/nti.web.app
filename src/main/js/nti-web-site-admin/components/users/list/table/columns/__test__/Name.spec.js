/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import { TestUtils } from '@nti/web-client';

import Name from '../Name';

const { tearDownTestClient, setupTestClient } = TestUtils;

const mockService = {
	resolveEntity: () => Promise.resolve({ displayName: 'User 1' }),
};

const onBefore = () => {
	setupTestClient(mockService);
};

const onAfter = () => {
	tearDownTestClient();
};

describe('Site admin user table name column test', () => {
	beforeEach(onBefore);
	afterEach(onAfter);

	test('Test for learners', () => {
		const colCmp = renderer.create(<Name
			item={{
				email: 'user@blah.com',
				getID: () => 'user'
			}}
			store={{
				filter: 'learners'
			}}/>);

		const tree = colCmp.toJSON();

		expect(tree).toMatchSnapshot();
	});

	test('Test for admins', () => {
		const colCmp = renderer.create(<Name
			item={{
				email: 'user@blah.com',
				getID: () => 'user'
			}}
			store={{
				filter: 'admins'
			}}/>);

		const tree = colCmp.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
