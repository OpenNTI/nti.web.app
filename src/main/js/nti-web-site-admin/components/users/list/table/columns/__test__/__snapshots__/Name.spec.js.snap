// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site admin user table name column test Test for admins 1`] = `
<div
  className="pathLink nti-link-to-path"
  to="#"
>
  <div>
    <div>
      <div>
        user@blah.com
      </div>
    </div>
  </div>
</div>
`;

exports[`Site admin user table name column test Test for learners 1`] = `
<div
  className="pathLink nti-link-to-path"
  to="#"
>
  <div>
    <div>
      <div>
        user@blah.com
      </div>
    </div>
  </div>
</div>
`;
